OneCommunityGlobal/HGNRest
# Rachana gzip compression

# Description
The website now includes gzip compression using the compression library. This middleware is configured to compress responses based on content type while considering streaming requirements and server load. Testers should verify that compression works correctly and evaluate overall server performance with and without compression enabled.

## Related PRS (if any):
This backend PR is NOT related to any other PR
…

## Main changes explained:

1. Added `"compression": "^1.7.5"` in the `package.json` file to include the compression library.  
2. Integrated gzip compression in `app.js` using `compression` middleware. The configuration includes:  
   - `level: 6`: Balanced compression level for performance and efficiency.  
   - `threshold: 0`: Ensures all responses are compressed regardless of size.  

These changes optimize response sizes to improve application performance and reduce bandwidth usage.
…

## How to test:
Dont forget to npm install before building and running the application.

1. Run the application on `localhost`.  
2. Before logging in, open the browser Developer Tools by pressing **Ctrl + Shift + I**.  
3. Switch to the **Network** tab in Developer Tools.  
4. Login to the application.  
5. Observe the requests appearing in the **Network** tab (e.g., `{i} users` or `{i} login`).  
6. Click on any request to inspect its details.  
7. Under the **Headers** tab, locate the **Content-Encoding** field. It should display `"gzip"`, indicating compression is applied.  

## Screenshots or videos of changes:

https://github.com/user-attachments/assets/f441097a-7421-46c4-8250-19c97bdd20d7

## Note:
Include the information the reviewers need to know.
